### ISSUE TITLE
Rooms reserving 1 source rooms 2 steps away instead of 2 source room with low distnace sources right next door

### ISSUE THREAD

**Issue by Somotaw** [TooAngel/screeps#88]
Maybe change to be less random and focus on scouting and reserving the closest room with 2 sources first? Ideas?

**Comment by Somotaw** [comment]
![image](https://cloud.githubusercontent.com/assets/6491698/21668334/13135284-d301-11e6-9740-96d92d6b2a35.png)

**Comment by Somotaw** [comment]
perhaps let pathcost from sources to baseroom storage can be something to use?

**Comment by WhiteHalmos** [comment]
Hmm... not sure which one is yours, but reservations should only happen to adjacent rooms now, adjacent to a reserved room or owned room.

**Comment by TooAngel** [comment]
@WhiteHalmos I'm not sure about that. My implementation was somehow random, if a room is seen and can be reserved (too less reserved ones), it will be reserved. So there are cases where a couple of rooms away are reserved, not likely, but possible.

My current idea is a regular reservation check, especially if players near you vanish or newbie walls open up. And do if more in a BFS manner. Doesn't have any concrete ideas yet, except of BFS and regular.

**Comment by WhiteHalmos** [comment]
I added code to only reserve rooms for which all the rooms along the path must be reserved or owned: https://github.com/TooAngel/screeps/commit/a96cf48bb8b3fe218213fb6bf2b16d0f79dc0b50

**Comment by WhiteHalmos** [comment]
+1 for regular check, maybe just keep spawning scouts, they're super cheap anyways.

**Comment by TooAngel** [comment]
IMHO the scouts are the pre observer solution. With observers it is different.

Thought about storing the number of sources in room memory, so that you are able to calculate the best rooms to reserve based on the info from memory.

**Comment by Somotaw** [comment]
we should store their positions, so we can calculate full pathcost/source to determine remote mining efficiency? :)

**Comment by TooAngel** [comment]
Good point, so instead of storing the number of sources we can store a list of source IDs.

The path length can be calculated by finding the route and following the paths from `storage` to `ID`.

**Comment by Somotaw** [comment]
Yes this sounds good, adding to todo :)
